File: SEGUNDO/C1/NetBeansProjects/Deep_space/src/deepspace/SuppliesPackage.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;

/**
 *
 * @author daniel
 */
public class SuppliesPackage {
    private float ammoPower;
    private float fuelUnits;
    private float shieldPower;
    
    SuppliesPackage(float ammopower, float fuelunits, float shieldpower){
        ammoPower=ammopower;
        fuelUnits=fuelunits;
        shieldPower=shieldpower;
    }
    
    SuppliesPackage(SuppliesPackage s){
        this(s.ammoPower,s.fuelUnits,s.shieldPower);
    }

    public float getAmmoPower() {
        return ammoPower;
    }

    public float getFuelUnits() {
        return fuelUnits;
    }

    public float getShieldPower() {
        return shieldPower;
    } 
    
    public void to_String(){
        System.out.println("Munición: " + ammoPower + "\n");
        System.out.println("Combustible: " + fuelUnits + "\n");
        System.out.println("Escudo: " + shieldPower + "\n");
    }
}

File: SEGUNDO/C2/PDOO/backup/DeepSpace_ruby/pkg/DeepSpace_ruby-0.0.1/lib/SuppliesPackage.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
#encoding:utf-8
module Deepspace
  
class SuppliesPackage
  attr_reader :ammoPower, :fuelUnits, :shieldPower
  
  def initialize(ammoPower,fuelUnits,shieldPower)
    @ammoPower = ammoPower
    @fuelUnits = fuelUnits
    @shieldPower = shieldPower    
  end
  
  def self.newCopy(copia)
    new(copia.ammoPower, copia.fuelUnits,copia.shieldPower)
  end
  
  def to_s
    return "Poder de Municion: #{@ammoPower}\nUnidades de Combustible: #{@fuelUnits}\nPoder de Escudo: #{@shieldPower}\n"
  end

end

end
